src/App.jsx

```diff
@@ -1,10 +1,112 @@
 import './App.css';
+import {useState} from "react";
+import axios from "axios";
+import countryColorName from "./helper/countryColorName.jsx";
+import RoundPopulation from "./helper/RoundPopulation.jsx";
+import worldMap from "./assets/world_map.png";
+
 
 function App() {
 
+    const [loading, setLoading] = useState(false);
+    const [error, setError] = useState(false);
+    const [allCountries, setAllCountries] = useState([]);
+    const [buttonVisability, setButtonVisability] = useState(true);
+    const [singleCountry, setSingleCountry] = useState(null);
+    const[searchCountry, setSearchCountry] = useState("");
+    const [currentPage, setCurrentPage] = useState(1);
+
+    const handleAllCountries = async () => {
+
+        try {
+            setLoading(true)
+            const response = await axios.get('https://restcountries.com/v3.1/all');
+            let sortAllCountries = [...response.data].sort((a, b) => {
+                return a.population - b.population;
+            })
+            setAllCountries(sortAllCountries);
+            console.log(response.data);
+
+        } catch (err) {
+            setError(err.message);
+console.error(err);
+
+    } finally  {
+            setButtonVisability(false);
+            setLoading(false);
+    }}
+
+    const handleSingleCountry = async () => {
+        setSingleCountry(null);
+        setSearchCountry(searchCountry);
+        setError(null)
+
+
+        try{
+            setLoading(true)
+            const response = await axios.get(`https://restcountries.com/v3.1/name/${searchCountry}`);
+            setSingleCountry(response.data[0]);
+            console.log(response.data[0]);
+        } catch (err) {
+            setError(err.message);
+            console.error(err)
+
+        } finally {
+            setSearchCountry(" ")
+            setLoading(false);
+        }
+    }
+
     return (
         <>
-            Maak je applicatie hier!
+            <span className="world-map">
+            <img src={worldMap} alt="WorldMap" />
+            </span>
+            <div className="buttons">
+            {buttonVisability && (
+                <div>
+                    <button onClick={handleAllCountries}>All Countries</button>
+                </div>
+            )}
+
+            <label htmlFor="search-country">Search single country:
+                <input type="search" id="search-country" value={searchCountry} name="q"
```

Comment:
Als je een input hebt is het beter om een form element te gebruiken met een onSubmit.